### ISSUE TITLE
Ability to define camel-case rules more flexibly?

### ISSUE THREAD

**Issue by richardgirges** [eslint/eslint#6948]
I'm trying to setup some lint rules around camel casing in a universal JS app. However, I've hit a wall with (what appears to be) lack of flexibility around the `camelcase` rule in eslint.

Apologies if there is a way to do what I'm asking - I've been searching around for a couple hours but haven't found a clear path to achieve what I'm trying to do. 

What I'm trying to achieve is the following set of rules:
* All variable and function names must be camel case
* JS object properties can be snake case 
* dot notation properties can be snake case

Example:
```javascript
var myVariable = 'foo'; // valid
var your_variable = 'bar'; // invalid

// valid
function helloWorld() {
}

// invalid
function hello_world() {
}

var myObject = {
  hello_world: 'foo', // valid
};

var yourObject = {};
yourObject.hello_world = 'foo'; // valid
```

The reason being that we're working with a MySQL database that contains table names and fields in snake case format. So we find ourselves needing to do things like this often:

```javascript
executeApiCallThatUpdatesDatabase({
  'user_id': 1234,
  'email': 'foo@bar.com'
});
```

If we were able to apply camel case rules only for var/function declarations, I believe that would solve the problem. Open to any other suggested routes to achieving this result. Thanks!

**What version of ESLint are you using?**
v3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Comment by richardgirges** [comment]
And just like that after posting this up I figured it out. This is what I needed:

`camelcase: ["error", {"properties": "never"}]`

Didn't realize that the `properties` option in the `camelcase` rule is specifically targeting object properties. I'm a dummy. Nothing to see here, move along.
